### DISCUSSION TITLE
Player.options(obj) Doesn't Update the Rendered Player

### DISCUSSION THREAD

**Discussion by kecoliva** [videojs/video.js#8371] [Q&A]
I have an instance of the video.js player and I'm trying to update its options using `Player.options(obj)` function but it doesn't seem to update the rendered player itself. But upon checking when I log `Player.options_` I can see that the specific option I updated (e.g muted) is updated but the rendered video player doesn't seem to reflect these changes.

**Comment by mister-ben** [comment]
It's not intended that that would update the player. While changing some options would have an effect if updated because the value is read from options_ after the change, many are only read when the player / a component initialises.

For muted, use the `player.muted(true)` setter.

**Reply by kecoliva** [reply]
@mister-ben If that's the case, is there any way to update the player's options so that the changes would appear in the video player component itself? I'm trying to make my options dynamic and not just changing muted status of the player component.

**Reply by mister-ben** [reply]
You could update options on event listeners, e.g.
```js
player.on('volumechange', function() {
  this.options({
     muted: this.muted(),
     volume: this.volume()
  });
});
```

**Reply by kecoliva** [reply]
Interesting... Can I achieve this outside an event listener? Similar to what I first expected how Player.options(obj) would work.

**Reply by kecoliva** [reply]
@mister-ben I tried the code you shared above and it doesn't update the video's options.
